src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copy = {};
+
+  Object.assign(copy, state);
+
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copy, action.extraData);
+      result.push({ ...copy });
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete copy[key];
+      }
+      result.push({ ...copy });
+    }
+
+    if (action.type === 'clear') {
```

Comment:
Consider using a switch statement instead of multiple if statements for handling 'action.type'. This is recommended when you have a limited number of conditions.